Transition to govuk-frontend

jonheslop opened the pull request:
The product page example was made a while ago and uses the now deprecated `govuk-elements-sass, govuk_frontend_toolkit, govuk_template_jinja` this updates the app to use `govuk-frontend`. 

This will also help users meet WCAG 2.0

36degrees commented on config.rb:
```diff
@@ -49,10 +49,26 @@
+# Because govuk-frontent fonts and images are with node_modules so they aren’t _seen_ by sprockets
+fonts_dir = Dir[File.join(root, "node_modules/govuk-frontend/assets/fonts/*")]
+fonts_dir.each do |filename|
+  dest_folder = File.join(root, "source/assets/fonts")
```
This falls over if the font doesn't already exist (e.g. on a clean install)

jonheslop replied:
Ah good point, will do a `.gitkeep` on the folders

nickcolley commented on config.rb:
```diff
@@ -48,7 +48,7 @@
 configure :build do
   # Minify CSS on build
-  activate :minify_css
+  # activate :minify_css
```
There's a chance that it's related to https://github.com/alphagov/govuk-frontend/issues/1350

jonheslop replied:
Yeah its the same problem

freesteph commented:
We need to update the [README.md's Components section](https://github.com/alphagov/product-page-example/tree/transition-to-govuk-frontend#components) and remove those who don't have their own SASS anymore (Phase Banner would be one of them).

freesteph commented on source/layouts/layout.erb:
```diff
@@ -2,68 +2,47 @@
-          <nav id="navigation" class="header__navigation js-nav" aria-label="Top Level Navigation" data-click-events data-click-category="Header" data-click-action="Navigation link clicked">
-            <ul>
+        <div class="govuk-header__content">
+          <button role="button" class="govuk-header__menu-button js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
```
I can't see this button anywhere, I suspect a call is missing to [initAll()](https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md#option-1-include-javascript) as I can't find a grep match for it

jonheslop replied:
Ah good spot, I have also missed these two things "Update CSS class names" and "Update and add data-module attributes" from https://github.com/alphagov/govuk-frontend/releases/tag/v3.0.0 which make the menu work

jonheslop replied:
I’ve got this working now

freesteph commented on config.rb:
```diff
@@ -52,38 +51,31 @@
-end
-
-# Override generated asset path to remove the long explicit vendor paths
-class ImportedAssetPathProcessor
```
we managed to keep this code and just use 

```
//= link_tree ../../node_modules/govuk-frontend/govuk/assets
```

in the Sprockets manifest, wouldn't this work here too?

jonheslop replied:
I am a ruby noob, if I did this could I delete my shonky copy stuff I added below, because if so this would be better I guess?

freesteph replied:
I'll try and provide you with a patch/commit, if not let's just roll with what works

freesteph replied:
Et voilà ! https://github.com/alphagov/product-page-example/pull/51 has the changes. It removes the need for a custom Sprockets processor so let me know what you think and no offence if deemed unnecessary 😄

jonheslop commented:
Ok I added a few more commits to sort out the JS.
- remove old JS
- added and init’d govuk-frontend’s JS
- added the right class names and data-module names and also the little bit of JS that adds `js-enabled` to `body`

I have deployed the built version if you want to see it in action https://build-lb4l4qks0.now.sh

freesteph commented on source/layouts/layout.erb:
```diff
@@ -1,69 +1,59 @@
-                  'Supplementary Pages' => '/supplementary.html',
-                  'Support' => '/support.html'
+                  'Supplementary Pages' => '/supplementary',
+                  'Support' => '/support'
```
This active state logic only works for the first "About" link because the `current_page.url` includes the trailing slash. I've added the slashes locally and it does fix the problem.

I don't know how to do the GitHub suggestion-comment thing so here's a patch if you want to use that:

```patch
From fe6d950cb40498be753e386c28c87b1c8dd5c8c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Maniaci?=
 <stephane.maniaci@digital.cabinet-office.gov.uk>
Date: Wed, 28 Aug 2019 14:25:12 +0100
Subject: [PATCH] Fix the active menu item logic

`current_page.url` includes the trailing slash so it needs to be
present in the menu map as well.
---
 source/layouts/layout.erb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/layouts/layout.erb b/source/layouts/layout.erb
index d0402bb..bbcd6b4 100644
--- a/source/layouts/layout.erb
+++ b/source/layouts/layout.erb
@@ -48,8 +48,8 @@
               <%
                 {
                   'About' => '/',
-                  'Supplementary Pages' => '/supplementary',
-                  'Support' => '/support'
+                  'Supplementary Pages' => '/supplementary/',
+                  'Support' => '/support/'
                 }.each do |title, url|
               %>
                 <li class="govuk-header__navigation-item<% if url == current_page.url %> govuk-header__navigation-item--active" aria-current="page<% end %>">
-- 
2.17.2 (Apple Git-113)
```

jonheslop replied:
ah good spot
